### PR TITLE
Listing of files and fetching files

### PR DESCRIPTION
**Pull request by likp**
This PR contains implementations for 
- Listing files in a knowledge library
- Fetching a file from a knowledge library

### CODE REVIEW

File: src/Balsam/src/Balsam.Api/Controllers/KnowledgeLibraryController.cs [hunk 1]
```diff
@@ -30,14 +34,56 @@ public KnowledgeLibraryController(ILogger<ProjectController> logger, HubClient h
...
+                }
+
+                Response.Headers.Add("content-disposition", "inline");
+
+                // Return the file stream
+                return File(stream, contentType);
+            }
+            catch (Exception ex)
+            {
+                // Log the exception here
```
Comment (scbamse) [inline src/Balsam/src/Balsam.Api/Controllers/KnowledgeLibraryController.cs:64]: Ha med en logg här för exceptionet

File: src/specs/Balsam.yaml [hunk 2]
```diff
@@ -377,7 +377,7 @@ paths:
     get:
       tags: 
         - KnowledgeLibrary
-      operationId: ListKnowledgeLibaryFiles
+      operationId: ListKnowledgeLibraryFiles
       description: List all files for a knowledge library
```
Comment (scbamse) [inline src/specs/Balsam.yaml:381]: Samma description som för ListKnowledgeLibraryFileContent

File: src/Balsam/src/Balsam.Api/Controllers/KnowledgeLibraryController.cs [hunk 3]
```diff
@@ -30,14 +34,56 @@ public KnowledgeLibraryController(ILogger<ProjectController> logger, HubClient h
...
+            try
+            {
+                string filePath = string.Empty;
+                filePath = _knowledgeLibraryClient.GetRepositoryFilePath(libraryId, fileId);
+
+                // Open the file
+                var stream = System.IO.File.OpenRead(filePath);
+
+                // Get the file extension
+                var extension = System.IO.Path.GetExtension(filePath);
```
Comment (paah-scb) [inline src/Balsam/src/Balsam.Api/Controllers/KnowledgeLibraryController.cs:48]: `extension ` never used

File: src/specs/Balsam.yaml [hunk 4]
```diff
@@ -400,7 +400,7 @@ paths:
     get:
       tags: 
         - KnowledgeLibrary
-      operationId: ListKnowledgeLibaryFileContent
+      operationId: ListKnowledgeLibraryFileContent
       description: List all files for a knowledge library
```
[the remark this answers is not in the document]
Comment (likp) [reply src/specs/Balsam.yaml]: Updated the comment and change the name to GetKnowledgeLibraryFileContent

File: src/specs/Balsam.yaml (hunk 2 above)
Comment (likp) [reply src/specs/Balsam.yaml:381]: See below

File: src/Balsam/src/Balsam.Api/Controllers/KnowledgeLibraryController.cs (hunk 1 above)
Comment (likp) [reply src/Balsam/src/Balsam.Api/Controllers/KnowledgeLibraryController.cs:62]: Added logging

**Review by paah-scb — APPROVED** [review]

**Review by scbamse — APPROVED** [review]

**Pull request merged**